packages/core/src/mask/MaskData.js

```diff
@@ -0,0 +1,115 @@
+import { MASK_TYPES } from '@pixi/constants';
+
+/**
+ * Component for masked elements
+ *
+ * Holds mask mode and temporary data about current mask
+ *
+ * @class
+ * @memberof PIXI
+ */
+export class MaskData
+{
+    /**
+     * Create MaskData
+     *
+     * @param {PIXI.DisplayObject} [maskObject=null] object that describes the mask
+     */
+    constructor(maskObject)
+    {
+        /**
+         * Mask type
+         * @member {PIXI.MASK_TYPES}
+         */
+        this.type = MASK_TYPES.NONE;
+
+        /**
+         * Whether we know the mask type beforehand
+         * @member {boolean}
+         * @default true
+         */
+        this.autoDetect = true;
+
+        /**
+         * Which element we use to mask
+         * @member {PIXI.DisplayObject}
+         */
+        this.maskObject = maskObject || null;
+
+        /**
+         * Whether it belongs to MaskSystem pool
+         * @member {boolean}
+         */
+        this.pooled = false;
+
+        /**
+         * Indicator of the type
+         * @member {boolean}
+         */
+        this.isMaskData = true;
+
+        /**
+         * Stencil counter above the mask in stack
+         * @member {number}
+         * @private
+         */
+        this._stencilCounter = 0;
+        /**
+         * Scissor counter above the mask in stack
+         * @member {number}
+         * @private
+         */
+        this._scissorCounter = 0;
+
+        /**
+         * Scissor operation above the mask in stack.
+         * Null if _scissorCounter is zero, rectangle instance if positive.
+         * @member {PIXI.Rectangle}
+         */
+        this._scissorRect = null;
+
+        /**
+         * Targeted element. Temporary variable set by MaskSystem
+         * @member {PIXI.DisplayObject}
+         * @private
+         */
+        this._target = null;
+    }
+
+    /**
+     * resets the mask data after popMask()
+     */
+    reset()
+    {
+        if (this.pooled)
+        {
+            this.maskObject = null;
+
+            this.type = MASK_TYPES.NONE;
+
+            this.autoDetect = true;
+        }
+
+        this._target = null;
+    }
+
+    /**
+     * copies counters from maskData above, called from pushMask()
+     * @param {PIXI.MaskData|null} maskAbove
+     */
+    copyCounters(maskAbove)
```

Comment:
I see you hide implementation details for that method, so `copyCountersOrReset`?
Or if there are no maskAbove, that means the mask is the entire screen?